crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_comprehension_in_call.rs

```diff
@@ -82,7 +92,7 @@ pub(crate) fn unnecessary_comprehension_any_all(
     let Expr::Name(ast::ExprName { id, .. }) = func else {
         return;
     };
-    if !matches!(id.as_str(), "all" | "any") {
+    if !matches!(id.as_str(), "all" | "any" | "sum" | "min" | "max") {
         return;
     }
```

Comment:
It's unrelated to your fix, so feel free to ignore, but possibly this could be

```rs
    if !checker
        .semantic()
        .resolve_qualified_name(func)
        .is_some_and(|qualified_name| {
            matches!(
                qualified_name.segments(),
                ["" | "builtins", "all" | "any" | "sum" | "min" | "max"]
            )
        })
    {
        return;
    }
```

And then the rule would also trigger if a user did something like this:

```py
import builtins

builtins.sum([2 for x in range(100)])
```

as well as this:

```py
sum([2 for x in range(100)])
```

A bit of an edge case, though, since obviously importing `builtins` is pretty rare

Reply:
I think the important thing here is that we don't trigger if the user shadows `any` with their own function. And we don't, because there is a `checker().semantic().is_builtin(id)` check below.

I don't think we need to care about `builtins.sum(...)`, nobody does that :)

Reply:
> I don't think we need to care about `builtins.sum(...)`, nobody does that :)

Well, [some people do](https://grep.app/search?q=builtins.sum%28&filter[lang][0]=Python), but I agree it's v rare :)

Reply:
Ok, point taken :) Since I'll need to update this to move it to preview anyway, I may integrate your suggestion.